Azure-Samples/azure-ml-federated-learning
# Share resources and standardize component names

## Purpose

This PR bundles a few different changes
* use the shared aggregation component across all examples
* use a single {requirements, environment} doublet across all examples (and CI/CD)
* standardize component spec names to spec.yaml
* upgrade old pip version in components

Successful jobs:
- [literal MNIST](https://ml.azure.com/experiments/id/b7743d38-84d1-4ff2-b2c3-6ae57147ea71/runs/clever_spinach_d9sgz36s1s?wsid=/subscriptions/48bbc269-ce89-4f6f-9a12-c6f91fcb772d/resourcegroups/thopo-owner-fldev-rg/workspaces/aml-thopo-fl30&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)
- [literal HELLOWORLD](https://ml.azure.com/runs/cool_turnip_djxm20v3ph?wsid=/subscriptions/48bbc269-ce89-4f6f-9a12-c6f91fcb772d/resourcegroups/thopo-owner-fldev-rg/workspaces/aml-thopo-fl30&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)
- [factory MNIST](https://ml.azure.com/runs/patient_onion_s4s7b3q6gk?wsid=/subscriptions/48bbc269-ce89-4f6f-9a12-c6f91fcb772d/resourcegroups/thopo-owner-fldev-rg/workspaces/aml-thopo-fl30&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)
- [factory HELLOWORLD](https://ml.azure.com/runs/sincere_cumin_kmdbgnjfr5?wsid=/subscriptions/48bbc269-ce89-4f6f-9a12-c6f91fcb772d/resourcegroups/thopo-owner-fldev-rg/workspaces/aml-thopo-fl30&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)
- [PNEUMONIA](https://ml.azure.com/runs/quiet_pipe_68m94pf3w9?wsid=/subscriptions/48bbc269-ce89-4f6f-9a12-c6f91fcb772d/resourcegroups/thopo-owner-fldev-rg/workspaces/aml-tp-pneumo&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

Closes #173 #174 #180 

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Get the code
* Submit all example jobs (already done, see successful job links above)
